src/com/amazon/awslabs/cloudfoody/OpenTable.java

```diff
@@ -13,47 +13,111 @@
+        Random myRandomizer = new Random();
+        return restaurants.get(myRandomizer.nextInt(restaurants.size()));
+    }
 
+    /**
+     * Get restaurants by city from Unofficial Open Table API.
+     * @param city
+     * @param state
+     * @return ArrayList of restaurants
+     * @throws IOException
+     */
+    public static ArrayList<Restaurant> getRestaurantsInCity(String city, final String state) throws IOException {
+        int pageNumber = getRandomPage(city, state);
+        String fullURL = BASE_URL + "restaurants?city=" + replaceSpacesWithPlus(city) + "&state=" + state + "&page=" + pageNumber + "&per_page=100";
         URLConnection connection = createConnection(fullURL);
 
         return parseRestaurants(connection.getInputStream());
     }
 
     /**
-     * Get restaurants by city from Unofficial Open Table API. Limited to the first 100 results.
-     * @param city city name
-     * @return a list of restaurants
+     * Get a random page number for multiple pages results
+     * @param city
+     * @param state
+     * @return int pageNumber
      * @throws IOException
      */
-    public static ArrayList getRestaurantsInCity(final String city) throws IOException {
-        String fullURL = BASE_URL + "restaurants?city=" + city + "&per_page=100";
+    private static int getRandomPage(String city, final String state) throws IOException {
+        String fullURL = BASE_URL + "restaurants?city=" + replaceSpacesWithPlus(city) + "&state=" + state;
         URLConnection connection = createConnection(fullURL);
+        HashMap resultsHash = new ObjectMapper().readValue(new InputStreamReader(connection.getInputStream()), HashMap.class);
+
+        /* Get the total number of restaurants at this city then divide is be maximum number of restaurants per page with is 100
+           to get the number of pages.
+         */
 
+        int totalRestaurants = (int) resultsHash.get("total_entries");
+        int numberOfPages = (int) Math.ceil(totalRestaurants/100.00);
+
+        if (numberOfPages == 1) {
+            return 1;
+        } else {
+            return ThreadLocalRandom.current().nextInt(1, numberOfPages + 1);
+        }
+    }
+
+    /**
+     * Get restaurants by name from Unofficial Open Table API.
+     * @param name restaurant name
+     * @return a list of restaurants
+     * @throws IOException
+     */
+    public static ArrayList<Restaurant> getRestaurantsByName(String name) throws IOException {
+        String fullURL = BASE_URL + "restaurants?name=" + replaceSpacesWithPlus(name);
+        URLConnection connection = createConnection(fullURL);
         return parseRestaurants(connection.getInputStream());
     }
 
     /**
-     *  Establish a connection and return a url object.
+     * Check if a string contains spaces and replace them with pluses
+     * @param originalString
+     * @return String
+     */
+    private static String replaceSpacesWithPlus(final String originalString) {
```

Comment:
You won't need this helper method if you encode URL with existing utilities (e.g. URLEncoder).